torch/csrc/dynamo/init.cpp

```diff
@@ -14,6 +15,22 @@ namespace torch {
 namespace dynamo {
 using torch::dynamo::autograd::torch_c_dynamo_compiled_autograd_init;
 
+namespace {
+
+// An RAII to save and restore the current local dispatch keyset. Shouldn't
+// normally be needed - mostly for emergency use.
+struct PreserveDispatch {
+  c10::impl::LocalDispatchKeySet m_old;
+
+  ~PreserveDispatch() {
+    c10::impl::_force_tls_local_dispatch_key_set(m_old);
+  }
+
+  PreserveDispatch() : m_old(c10::impl::tls_local_dispatch_key_set()) {}
+};
```

Comment:
Could you move this to `torch/csrc/utils/python_dispatch.cpp`? That's where our other utilities go.

Reply:
Also, nit: please change the naming to include the word "Guard" (see the other examples in python_dispatch.cpp)